### PR TITLE
refactor: optimised fetching key from keystore dir

### PR DESCRIPTION
**Pull request by Yashk767**
# Description

Previously, when fetching keys from the keystore, we used ks.Accounts(), which created a new goroutine every time. This approach led to goroutines waiting indefinitely, which was inefficient and could potentially cause resource exhaustion issues. To optimize this process and avoid unnecessary resource consumption, the solution is to use the same keystore instance every time to call ks.Accounts() will solves the issue of increasing go routines.

So this PR optimises keystore file fetching to prevent indefinite waiting by goroutines.

Fixes https://github.com/razor-network/oracle-node/issues/1195

Solution
Creating and initialising the keystore instance once at the start and using the same to call `ks.Accounts()` at every fetching of pKey call doesnt increase the goroutine count and it also gives the updated keystore cache at the time of fetching accounts. 

How Has This Been Tested?
- Added realistic fetching of pKey from keystore file tests.
- Account Imports: Tested the creation and importation of both new and old accounts to verify compatibility and functionality across different account types and ages.

### CODE REVIEW

**Comment by ashish10677** [comment]
Tests failing @Yashk767

File: accounts/accounts.go [hunk 1]
```diff
@@ -62,3 +68,44 @@ func (AccountUtils) SignData(hash []byte, account types.Account, defaultPath str
 	}
 	return AccountUtilsInterface.Sign(hash, privateKey)
 }
+
+// FindKeystoreFileForAddress matches the keystore file for the given address.
+func FindKeystoreFileForAddress(keystoreDirPath, address string) (string, error) {
+	normalizedAddress := strings.ToLower(address)
+	if strings.HasPrefix(normalizedAddress, "0x") {
+		normalizedAddress = normalizedAddress[2:]
```
Comment (ashish10677) [inline accounts/accounts.go:76]: @Yashk767 Please check this suggestion. It's the right way.

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: accounts/accounts.go (hunk 1 above)
Comment (Yashk767) [reply accounts/accounts.go:76]: yes, some tests are failing, fixing those. Will push it together.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -60,7 +60,7 @@ jobs:
 
       - name: Execute test case
         run: >
-          go-acc ./... --ignore razor/accounts/mocks --ignore razor/cmd/mocks
+          go-acc ./... --ignore razor/accounts/mocks --ignore razor/accounts/test_accounts --ignore razor/cmd/mocks
```
Comment (ashish10677) [inline .github/workflows/ci.yml:63]: Why have you excluded `test_accounts`?

Comment (Yashk767) [reply .github/workflows/ci.yml:63]: `test_accounts` directory has keystore files of testing addresses

**Review by ashish10677 — DISMISSED** [review]

**Comment by SkandaBhat** [comment]
`package accounts

import (
	"crypto/ecdsa"
	"github.com/ethereum/go-ethereum/accounts"
	"github.com/ethereum/go-ethereum/accounts/keystore"
	"razor/logger"
	"razor/path"
	"os"
	"strings"
)

type AccountManager struct {
	log *logger.Logger
	ks  *keystore.KeyStore
}

// NewAccountManager creates a new AccountManager instance.
func NewAccountManager() *AccountManager {
	return &AccountManager{
		log: logger.NewLogger(),
	}
}

// InitializeKeystore initializes the keystore instance.
func (am *AccountManager) InitializeKeystore(keystorePath string) {
	am.log.Info("Initialising keystoreInstance...")
	am.ks = keystore.NewKeyStore(keystorePath, keystore.StandardScryptN, keystore.StandardScryptP)
}

// CreateAccount creates a new account.
func (am *AccountManager) CreateAccount(keystorePath string, password string) accounts.Account {
	// Method implementation...
}

// GetPrivateKey retrieves the private key from the keystore.
func (am *AccountManager) GetPrivateKey(address string, password string, keystoreDirPath string) (*ecdsa.PrivateKey, error) {
	// Method implementation...
}

// SignData signs the data.
func (am *AccountManager) SignData(hash []byte, account types.Account, defaultPath string) ([]byte, error) {
	// Method implementation...
}
`

Please follow this pattern. cc: @ashish10677

**Pull request closed without merging**